src/Fetcher/ConfTechFetcher.php

```diff
@@ -13,117 +13,90 @@
-    }
+    private LocationGuesser $locationGuesser;
+    private HttpClientInterface $client;
+    private LoggerInterface $logger;
 
-    public function getUrl(array $params = []): string
+    public function __construct(LocationGuesser $locationGuesser, ?HttpClientInterface $client = null, ?LoggerInterface $logger = null)
     {
-        return "https://raw.githubusercontent.com/tech-conferences/conference-data/master/conferences/$params[date]/$params[tag].json";
+        $this->locationGuesser = $locationGuesser;
+        $this->client = $client ?: HttpClient::create();
+        $this->logger = $logger ?: new NullLogger();
     }
 
-    public function fetch(): \Generator
+    public function fetch(array $configuration = []): ?\Generator
     {
         foreach ([date('Y'), date('Y', strtotime('+1 year'))] as $date) {
-            foreach ($this->tagRepository->findBy(['selected' => true]) as $tag) {
-                $tagKey = array_search($this->slugger->slug(strtolower($tag->getName()))->toString(), self::TAGS_SYNONYMS);
+            if (0 === \count($configuration) || 0 === \count($configuration['tags'])) {
+                $this->logger->notice('This fetcher is not configured and fetches all data from the source. If this behaviour is not intended, you should add a configuration in the admin');
 
-                if (false === $tagKey) {
-                    continue;
-                }
+                foreach (self::SOURCE_AVAILABLE_TAGS as $tag) {
+                    $url = sprintf('https://github.com/tech-conferences/conference-data/tree/main/conferences/%s/%s.json', $date, $tag);
 
-                $tagSynonym = self::TAGS_SYNONYMS[$tagKey];
-                $url = $this->getUrl(['date' => $date, 'tag' => $tagSynonym]);
-
-                try {
-                    $response = $this->httpClient->request('GET', $url);
-                } catch (TransportExceptionInterface $exception) {
-                    $this->logger->error('HttpClient Transport Exception', [
-                        'url' => $url,
-                        'source' => self::SOURCE,
-                        'exception' => $exception->getMessage(),
-                    ]);
-                    continue;
+                    foreach ($this->queryConfTech($url) as $conference) {
+                        yield $this->denormalizeConference($conference);
+                    }
                 }
 
-                if (404 === $response->getStatusCode()) {
-                    $this->logger->error('Source URL returns 404', [
-                        'url' => $url,
-                        'source' => self::SOURCE,
-                    ]);
-                    continue;
-                }
+                return;
+            }
 
-                $data = $response->toArray();
+            foreach ($configuration['tags'] as $tag) {
+                $url = sprintf('https://github.com/tech-conferences/conference-data/tree/main/conferences/%s/%s.json', $date, $tag);
```

Comment:
This url should be stored in a private constant, no?

Reply:
Up to @HedicGuibert ... To me: there are no real values to create a constant for that... but not a big deal.

Anyway, I want to @HedicGuibert rewrite this fetcher with a `git clone`

Reply:
Oh, with a GitHub API Client? It's a very nice idea :+1: